lib/src/model/robotoff_nutrient_extraction.dart

```diff
@@ -0,0 +1,201 @@
+
+  factory RobotoffNutrientExtractionResult.fromJson(
+          Map<String, dynamic> json) =>
+      _$RobotoffNutrientExtractionResultFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() =>
+      _$RobotoffNutrientExtractionResultToJson(this);
+}
+
+@JsonSerializable()
+class NutrientEntity {
+  final int? start;
+  final int? end;
+  final String? text;
+  final String? unit;
+  final double? score;
+  final bool? valid;
+  final String? value;
+  final String? entity;
+  @JsonKey(name: 'char_start')
+  final int? charStart;
+  @JsonKey(name: 'char_end')
+  final int? charEnd;
+
+  const NutrientEntity({
+    this.start,
+    this.end,
+    this.text,
+    this.unit,
+    this.score,
+    this.valid,
+    this.value,
+    this.entity,
+    this.charStart,
+    this.charEnd,
+  });
+
+  factory NutrientEntity.fromJson(Map<String, dynamic> json) =>
+      _$NutrientEntityFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientEntityToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotationData {
+  final String? unit;
+  final String? value;
+
+  const NutrientAnnotationData({
+    this.unit,
+    this.value,
+  });
+
+  factory NutrientAnnotationData.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationDataToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotation {
+  final Map<String, NutrientAnnotationData>? nutrients;
+  @JsonKey(name: 'serving_size')
+  final String? servingSize;
+  @JsonKey(name: 'nutrition_data_per')
+  final String? nutritionDataPer;
+
+  const NutrientAnnotation({
+    this.nutrients,
+    this.servingSize,
+    this.nutritionDataPer,
+  });
+
+  factory NutrientAnnotation.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationToJson(this);
+}
+
+@JsonSerializable()
+class NutrientDataWrapper {
+  final Map<String, List<NutrientEntity>>? entities;
+  final Map<String, NutrientEntity>? nutrients;
+  final NutrientAnnotation? annotation;
+  @JsonKey(name: 'was_updated')
+  final bool? wasUpdated;
+
+  const NutrientDataWrapper({
+    this.entities,
+    this.nutrients,
+    this.annotation,
+    this.wasUpdated,
+  });
+
+  factory NutrientDataWrapper.fromJson(Map<String, dynamic> json) =>
+      _$NutrientDataWrapperFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientDataWrapperToJson(this);
+}
+
+@JsonSerializable()
+class NutrientExtractionInsight extends JsonObject {
+  @JsonKey(name: 'id')
+  final String? insightId;
+  final String? barcode;
+  final String? type;
+  final NutrientDataWrapper? data;
+  final String? timestamp;
+  @JsonKey(name: 'completed_at')
+  final String? completedAt;
+  final int? annotation;
+  @JsonKey(name: 'annotated_result')
+  final int? annotatedResult;
+  @JsonKey(name: 'n_votes')
+  final int? nVotes;
+  final String? username;
+  final List<String>? countries;
```

Comment:
Localized countries? Country codes?

Reply:
This is the kind of values in the list : "en:france"
Do we have an object to wrap this into? Or do you require to add a comment stating the format?

Reply:
`"en:france"` is unfortunately somehow useless. `'fr'` is useful.
If you could convince the server to send https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2 codes instead that would be better. If it's not possible, at least add a comment.